packages/contracts/test/zrx_token.ts

```diff
@@ -1,60 +1,69 @@
 import { ZeroEx } from '0x.js';
+import { BlockchainLifecycle, devConstants, web3Factory } from '@0xproject/dev-utils';
 import { BigNumber } from '@0xproject/utils';
+import { Web3Wrapper } from '@0xproject/web3-wrapper';
 import * as chai from 'chai';
-import Web3 = require('web3');
+import * as Web3 from 'web3';
 
-import { Artifacts } from '../../util/artifacts';
-import { constants } from '../../util/constants';
-import { ContractInstance } from '../../util/types';
+import { constants } from '../util/constants';
 
 import { chaiSetup } from './utils/chai_setup';
+import { deployer } from './utils/deployer';
 
 chaiSetup.configure();
 const expect = chai.expect;
-const { Exchange, ZRXToken } = new Artifacts(artifacts);
-const web3: Web3 = (global as any).web3;
+const web3 = web3Factory.create();
+const web3Wrapper = new Web3Wrapper(web3.currentProvider);
+const blockchainLifecycle = new BlockchainLifecycle(devConstants.RPC_URL);
 
-contract('ZRXToken', (accounts: string[]) => {
-    const owner = accounts[0];
-    const spender = accounts[1];
+describe('ZRXToken', () => {
+    let owner: string;
+    let spender: string;
     let zeroEx: ZeroEx;
 
     let MAX_UINT: BigNumber;
 
-    let zrx: ContractInstance;
+    let zrx: Web3.ContractInstance;
     let zrxAddress: string;
 
-    beforeEach(async () => {
+    before(async () => {
+        const accounts = await web3Wrapper.getAvailableAddressesAsync();
+        owner = accounts[0];
+        spender = accounts[1];
         zeroEx = new ZeroEx(web3.currentProvider, {
-            exchangeContractAddress: Exchange.address,
             networkId: constants.TESTRPC_NETWORK_ID,
         });
-        zrx = await ZRXToken.new();
+        zrx = await deployer.deployAsync('ZRXToken');
         zrxAddress = zrx.address;
         MAX_UINT = zeroEx.token.UNLIMITED_ALLOWANCE_IN_BASE_UNITS;
     });
-
+    beforeEach(async () => {
+        await blockchainLifecycle.startAsync();
+    });
+    afterEach(async () => {
+        await blockchainLifecycle.revertAsync();
+    });
     describe('constants', () => {
         it('should have 18 decimals', async () => {
-            const decimals = new BigNumber(await zrx.decimals.call());
+            const decimals = new BigNumber(await zrx.decimals());
```

Comment:
Hm, these methods on the contract wrapper don't have `async` suffixes?

Reply:
They don't for now, but they will with the upcomming abi-gen PR